### PR TITLE
Support view column comments for DeltaLake

### LINKED ISSUE #17773: Add support for comments on VIEW columns for Delta Lake
See similar implementation in HIve

https://github.com/trinodb/trino/blob/3298d6dbe84a1a65fb0aebfcf8008ebcde661e3a/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L1454

### PR DESCRIPTION
**Pull request by reswqa**
## Description
Support view column comments for DeltaLake

## Additional context and related issues
This closes #17773

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Delta Lake
* Add support for commenting on view columns ({issue}`17773`)
```

### CODE REVIEW

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java [hunk 1]
```diff
@@ -1242,6 +1242,13 @@ public void setViewComment(ConnectorSession session, SchemaTableName viewName, O
         trinoViewHiveMetastore.updateViewComment(session, viewName, comment);
     }
 
+    @Override
+    public void setViewColumnComment(ConnectorSession session, SchemaTableName viewName, String columnName,
+            Optional<String> comment)
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1247]: Please unwarp.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java [hunk 2]
```diff
@@ -193,6 +193,27 @@ public void updateViewComment(ConnectorSession session, SchemaTableName viewName
...
+        ConnectorViewDefinition definition = TrinoViewUtil.getView(viewName, view.getViewOriginalText(), view.getTableType(), view.getParameters(), view.getOwner())
+                .orElseThrow(() -> new ViewNotFoundException(viewName));
+        ConnectorViewDefinition newDefinition = new ConnectorViewDefinition(
+                definition.getOriginalSql(),
+                definition.getCatalog(),
+                definition.getSchema(),
+                definition.getColumns().stream()
+                        .map(currentViewColumn -> columnName.equals(currentViewColumn.getName()) ? new ConnectorViewDefinition.ViewColumn(currentViewColumn.getName(), currentViewColumn.getType(), comment) : currentViewColumn)
+                        .collect(toImmutableList()),
+                comment,
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:210]: This is wrong. Please use `definition.getComment()` instead.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java [hunk 3]
```diff
@@ -193,6 +193,27 @@ public void updateViewComment(ConnectorSession session, SchemaTableName viewName
         replaceView(session, viewName, view, newDefinition);
     }
 
+    public void updateViewColumnComment(ConnectorSession session, SchemaTableName viewName, String columnName, Optional<String> comment)
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:196]: Please extract `io.trino.plugin.iceberg.catalog.hms.TrinoHiveCatalog#updateViewColumnComment` in a preparatory commit.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java [hunk 4]
```diff
@@ -152,7 +152,7 @@ protected boolean hasBehavior(TestingConnectorBehavior connectorBehavior)
                 return false;
 
             case SUPPORTS_COMMENT_ON_VIEW_COLUMN:
-                return false;
+                return true;
```
Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:155]: Please remove `SUPPORTS_COMMENT_ON_VIEW_COLUMN` from this class.

Comment (ebyhr) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java:155]: `TestDeltaLakeConnectorTest` runs only on file metastore and it doesn't cover other metastore (HMS & Glue). Please add test to `BaseConnectorSmokeTest`.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 1 above)
Comment (reswqa) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1247]: Do you means let the parameter `Optional<String> comment` of `io.trino.plugin.hive.TrinoViewHiveMetastore#updateViewColumnComment` should be changed to `String comment`? I just follow the `updateViewComment` method and the constructor for `ConnectorViewDefinition.ViewColumn` is also expects a `Optional<String>`.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java (hunk 2 above)
Comment (reswqa) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:210]: Good catch! 👍

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java (hunk 4 above)
Comment (reswqa) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java]: Thank you for your reminder! I am not familiar with the related codes. 🙇

**Comment by reswqa** [comment]
Thanks @ebyhr for the patient review, I have updated this according to your comments. 

There is only one comment that has not been resolved as I may not fully understand(see comment reply). Would you mind taking a look again in your free time, thanks~

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java:1247]: I mean putting all arguments in a single line.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java [hunk 5]
```diff
@@ -193,6 +193,27 @@ public void updateViewComment(ConnectorSession session, SchemaTableName viewName
...
+                .orElseThrow(() -> new ViewNotFoundException(viewName));
+
+        ConnectorViewDefinition definition = TrinoViewUtil.getView(viewName, view.getViewOriginalText(), view.getTableType(), view.getParameters(), view.getOwner())
+                .orElseThrow(() -> new ViewNotFoundException(viewName));
+        ConnectorViewDefinition newDefinition = new ConnectorViewDefinition(
+                definition.getOriginalSql(),
+                definition.getCatalog(),
+                definition.getSchema(),
+                definition.getColumns().stream()
+                        .map(currentViewColumn -> columnName.equals(currentViewColumn.getName()) ? new ConnectorViewDefinition.ViewColumn(currentViewColumn.getName(), currentViewColumn.getType(), comment) : currentViewColumn)
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/TrinoViewHiveMetastore.java:208]: This line isn't equal to the original method. Please don't change the implementation when extracting a method.

**Comment by reswqa** [comment]
Thanks for the review! I have updated this in the fix-up comments, PTAL again.

**Comment by ebyhr** [comment]
Please update a product test.

**Comment by reswqa** [comment]
> Please update a product test.

Sorry, I am not familiar with the code base. Could you please tell me which file it is in. Thanks!

**Comment by ebyhr** [comment]
Please take a look at `io.trino.tests.product.hive.TestComments`.

**Comment by ebyhr** [comment]
Please take a look at `testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeView.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDeltaLakeView
+        extends BaseTestDeltaLakeS3Storage
+{
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
```
Comment (reswqa) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeView.java:16]: TBH, I am not sure which group this should belong to, or if a new group needs to be introduced. 🤔

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeViewComments.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_DATABRICKS;
+import static io.trino.tests.product.TestGroups.DELTA_LAKE_OSS;
+import static io.trino.tests.product.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestDeltaLakeViewComments
```
Comment (reswqa) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeViewComments.java:13]: I haven't found an existing class suitable for this test. Furthermore, I have not found any product testing code for comment on view (not on view column) in `io.trino.tests.product.deltalake`. If it does not exist, perhaps we should also add one?

Comment (reswqa) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeViewComments.java]: Sure, It doesn't matter. I have dropped this commit. 😆

**Comment by ebyhr** [comment]
/test-with-secrets sha=3446761eb93cce77cc28773bc774a4c9409df152

**Review by ebyhr — APPROVED** [review]

**Pull request merged**